OpenRA.Mods.Common/Traits/BotModules/BaseBuilderBotModule.cs

```diff
@@ -245,6 +268,32 @@ void IBotTick.BotTick(IBot bot)
 				}
 			}
 
+			if (--checkBestResourceLocationTicks <= 0 && resourceLayer != null)
+			{
+				checkBestResourceLocationTicks = Info.CheckBestResourceLocationInterval;
+
+				Actor bestconyard = null;
+				var best = int.MinValue;
+
+				foreach (var conyard in constructionYardBuildings.Actors.Where(a => !a.IsDead))
```

Comment:
```suggestion
				foreach (var conyard in constructionYardBuildings.Actors)
```

just move the check to inside the function

Reply:
fixed